### PR TITLE
[v8.9] [ssreflect] Export more parsing witnesses.

### PR DESCRIPTION
**Pull request by ejgallego**
This is needed in order to serialize ssreflect programs properly,
similar to #6795.

Backport of #9070 , this should be OK even for a minor release; it allows 8.9 SerAPI users to serialize more ssr scripts.

### CODE REVIEW

**Review by gares — APPROVED** [review]

**Review by maximedenes — APPROVED** [review]

File: plugins/ssr/ssrparser.ml4 [hunk 1]
```diff
@@ -342,7 +342,7 @@ let interp_index ist gl idx =
 
 open Pltac
 
-ARGUMENT EXTEND ssrindex TYPED AS ssrindex PRINTED BY pr_ssrindex
+ARGUMENT EXTEND ssrindex PRINTED BY pr_ssrindex
```
Comment (gares) [inline plugins/ssr/ssrparser.ml4]: Why is this needed? I did not notice this before. The semantics is not the same if ssrindex has non trivial functions attached to it.

Comment (ejgallego) [reply plugins/ssr/ssrparser.ml4]: This is needed as to remove a warning "ignored TYPED AS" annotation.

Comment (ejgallego) [reply plugins/ssr/ssrparser.ml4]: Maybe the right thing to do is to remove the `mli` file? I dunno, in 8.10 certainly things are different now.

Comment (gares) [reply plugins/ssr/ssrparser.ml4]: ok then.

File: plugins/ssr/ssrast.mli [hunk 2]
```diff
@@ -140,6 +140,9 @@ type 'tac ssrhint = bool * 'tac option list
 type 'tac fwdbinders =
         bool * (ssrhpats * ((ssrfwdfmt * ast_closure_term) * 'tac ssrhint))
 
+type 'tac ffwbinders =
+  (ssrhpats * ((ssrfwdfmt * ast_closure_term) * 'tac ssrhint))
```
Comment (SkySkimmer) [inline plugins/ssr/ssrast.mli]: Doesn't this factor with the above `fwbinders` as `type 'a fwdbinders = bool * 'tac ffwbinders`?

**Comment by Zimmi48** [comment]
8.9.1 has been tagged without this PR
